app/models/metric.rb

```diff
@@ -3,7 +3,21 @@ class Metric < ApplicationRecord
 
   include Metric::Common
 
-  def self.reindex_table_name
-    "metrics_#{Time.now.utc.hour + 1}"
+  def self.table_name
+    Thread.current[:metric_table] || super
+  end
+
+  def self.with_table_for_time(hour = Time.now.utc.hour)
+    Thread.current[:metric_table] = reindex_table_name(hour)
+    yield
+    self
```

Comment:
Is there a reason you are returning `self` here?  Doesn't seem to be used anywhere, as far as I can tell.